etna/datasets/tsdataset.py

```diff
@@ -198,6 +199,24 @@ def _cast_segment_to_str(df: pd.DataFrame) -> pd.DataFrame:
         df.columns = pd.MultiIndex.from_frame(columns_frame)
         return df
 
+    @staticmethod
+    def _check_timestamp_type(df: pd.DataFrame):
```

Comment:
I don't think this should be a separate method. Let's put the warning logic into `_cast_index_to_datetime`.